src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs

```diff
@@ -17,13 +17,43 @@ public ProviderOwnershipCache(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
             _cache = cache;
         }
 
+        public async Task<Guid?> GetProviderForCourse(Guid courseId)
+        {
+            var cacheKey = GetApprenticeshipCacheKey(courseId);
+
+            if (!_cache.TryGetValue<Guid?>(cacheKey, out var providerId))
+            {
+                // Cosmos document doesn't store ProviderID on the apprenticeship but UKPRN
```

Comment:
Confused by this comment about `apprenticeships`...

Reply:
v2 stuff uses a GUID ProviderId for referencing the provider but the Cosmos model(s) use a UKPRN.

Reply:
But the method is `GetProviderForCourse(...)` - what's this got to do with apprenticeships?

Reply:
Copy/paste fail 🙈